cql3/statements/select_statement.cc

```diff
@@ -1144,6 +1147,38 @@ query::partition_slice indexed_table_select_statement::get_partition_slice_for_g
             }
 
             partition_slice_builder.with_ranges(clustering_restrictions->bounds_ranges(options));
+        } else if (has_token(_restrictions->get_partition_key_restrictions()->expression) && is_token_v2_index) {
+            auto token_ranges = _restrictions->get_partition_key_restrictions()->bounds_ranges(options);
+            std::vector<query::clustering_range> all_bounds;
+
+            for (auto& token_range : token_ranges) {
+                auto& start_ring_postition = token_range.start()->value();
+                auto& end_ring_position = token_range.end()->value();
```

Comment:
Is it guaranteed that the interval is bounded? why?

Reply:
In this particular place we know that we are using `token_restriction` (`has_token` check is essentially a `token_restriction` check - it is the only class handling those types of restrictions). 

`bounds_ranges` returned by `token_restriction` always have `start()` and `end()` present:

https://github.com/scylladb/scylla/blob/f8c7d899b40a01d238b8adb35b93e4756f43837c/cql3/restrictions/token_restriction.hh#L100-L102

Reply:
Ah! But I want to have dht::{min,max}imum_token killed (I don't want to be the one doing the killing, though) because they are redundant with interval open bounds.

So please add checks. It's dangerous to rely on the caller anyway.

Reply:
Fixed it in the amended commit.